ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -117,7 +128,7 @@
         switch (message) {
           case FULL_ACCOUNT_METADATA_CHANGE_MESSAGE:
             logger.trace("Start processing message={} for topic={}", message, topic);
-            readFullAccountAndUpdateCache(FULL_ACCOUNT_METADATA_PATH);
+            readFullAccountAndUpdateCache(FULL_ACCOUNT_METADATA_PATH, true);
             logger.trace("Completed processing message={} for topic={}", message, topic);
             break;
           default:
```

Comment:
Somewhat unrelated to this patch, but do we want to throw an exception when we get an unrecognized message. Would this stop future listener calls from happening?

Reply:
Would prefer to handle the exception within the method, because that is the place to deal with the notifier. Any reason to throw an exception? Upper logic to handle this?